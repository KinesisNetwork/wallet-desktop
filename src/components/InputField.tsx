import * as React from 'react'

export interface Props {
  id: string
  value: string
  label?: string
  icon?: string
  helpText?: string
  placeholder?: string
  type?: 'password' | 'text' | 'number'
  isDisabled?: boolean
  hasButton?: boolean
  isInvalidInput?: boolean
  onChangeHandler: (newValue: string) => any
}

export const InputField: React.SFC<Props> = ({
  value,
  label,
  onChangeHandler,
  type = 'text',
  id,
  helpText,
  placeholder,
  icon,
  hasButton,
  isDisabled,
  isInvalidInput
}) => (
    <div className={`field ${hasButton ? 'control is-expanded' : ''}`}>
      {label && <label className='label is-small'>{label}</label>}
      <div className={`control ${icon ? 'has-icons-left' : ''}`}>
        <input
          className={`input ${isInvalidInput ? 'is-danger' : ''}`}
          id={`input-${id}`}
          onChange={(ev) => onChangeHandler(ev.target.value)}
          type={type}
          placeholder={placeholder}
          value={value}
          disabled={isDisabled}
        />
        {icon && <span className='icon is-left has-text-grey-lighter'><i className={`fas ${icon}`} /></span>}
      </div>
      <p className='help is-link'>
        {helpText}
      </p>
    </div>
  )

InputField.displayName = 'InputField'
