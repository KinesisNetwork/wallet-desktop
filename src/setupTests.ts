import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
