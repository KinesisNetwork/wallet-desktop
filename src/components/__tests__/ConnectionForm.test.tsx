import { mount, shallow } from 'enzyme';
import * as React from 'react'

import { ConnectionForm } from '@components/ConnectionForm'
import { InputField } from '@components/InputField'
import '../../setupTests'

describe('ConnectionForm', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ConnectionForm
        addConnection={() => null}
        handleConnectionFormChange={() => null}
        horizonURL=''
        name=''
        networkPassphrase=''
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('renders a heading section', () => {
    const wrapper = shallow(
      <ConnectionForm
        addConnection={() => null}
        handleConnectionFormChange={() => null}
        horizonURL=''
        name=''
        networkPassphrase=''
      />
    )

    expect(wrapper.find('h1').text()).toEqual('Add New Connection')
    expect(wrapper.find('.icon').children()).toHaveLength(1)
    expect(wrapper.find('.icon').children().hasClass('fa-plug')).toBe(true)
  })

  it('should render a form with three InputFields and a button', () => {
    const wrapper = shallow(
      <ConnectionForm
        addConnection={() => null}
        handleConnectionFormChange={() => null}
        horizonURL=''
        name=''
        networkPassphrase=''
      />
    )

    const form = wrapper.find('form')

    expect(form.exists()).toBe(true)
    expect(form.children()).toHaveLength(4)
    expect(form.prop('onSubmit')).toHaveLength(1)
    expect(form.childAt(3).find('button').text()).toEqual('Add Connection')

    const inputField = form.find(InputField)
    expect(inputField).toHaveLength(3)
  })

  it('calls handleSubmit function on submit', () => {
    const mockHandleSubmit = jest.fn()
    const wrapper = shallow(
      <ConnectionForm
        addConnection={mockHandleSubmit}
        handleConnectionFormChange={() => null}
        horizonURL='asd'
        name='zxc'
        networkPassphrase='qwe'
      />
    )

    const button = wrapper.find('form')
    button.simulate('submit', { preventDefault: () => null })
    expect(mockHandleSubmit).toHaveBeenCalled()
  })
})
