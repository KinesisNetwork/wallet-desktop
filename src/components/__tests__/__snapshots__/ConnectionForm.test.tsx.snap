// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionForm renders correctly 1`] = `
<section
  className="section"
>
  <div>
    <span
      className="icon is-large"
    >
      <i
        className="fas fa-plug"
      />
    </span>
    <h1
      className="title is-4 heading primary-font"
    >
      Add New Connection
    </h1>
  </div>
  <form
    onSubmit={[Function]}
  >
    <InputField
      id="connection-name"
      label="Connection Name"
      onChangeHandler={[Function]}
      value=""
    />
    <InputField
      id="connection-horizon-url"
      label="Horizon URL"
      onChangeHandler={[Function]}
      value=""
    />
    <InputField
      id="connection-network-passphrase"
      label="Network Passphrase"
      onChangeHandler={[Function]}
      value=""
    />
    <div
      className="field"
    >
      <div
        className="control is-expanded"
      >
        <button
          className="button is-fullwidth"
          type="submit"
        >
          Add Connection
        </button>
      </div>
    </div>
  </form>
</section>
`;
